runtime/client_http_request.go

```diff
@@ -165,7 +172,12 @@ func (req *ClientHTTPRequest) WriteJSON(
 	}
 
 	if body != nil {
-		httpReq.Header.Set("Content-Type", "application/json")
+		httpReq.Header.Set(http.CanonicalHeaderKey(contentTypeHeader), contentTypeHeaderValue)
```

Comment:
you may be overwriting the ContentType if its already present. I think you want to check first and set it only if it isn't set already (sort of same logic as accept header below).

Reply:
I also just noticed `req.defaultHeaders` above.  Why aren't we using default headers to set these default values? If the caller set the header, the loop on line 170 will overwrite the default header which is what you want anyway.

Reply:
That's modifying existing logic - outside the scope of this task.  

Also, Content-Type is to tell the downstream how to unmarshal the body - which in this case will always be application/json as long as there is a body.

Reply:
The default headers have to be specified using the config files, which individual endpoints have to set. In this case, we want to apply this to all endpoints since this is a platform level logic (always be application/json since unmarshaler is always application/json)

Reply:
What if the incoming request had `vnd+application/json` or something like for content-type? does the downstream not care to know that?

The current implementation requires you to have config per client, but can very easily be extended to use a default global config for all clients as well as config for the said client so instead of one config, using two config properties, is that really that much of scope creep?